### PR TITLE
fix(basic-auth): split schema to adapt different occasions

### PR DESCRIPTION
**Pull request by spacewander**
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Comment by liuhengloveyou** [comment]
fix which issues?

**Comment by spacewander** [comment]
The original schema check is too floppy. It will let the invalid schema in `TEST 10: invalid schema, not field given` pass, but when you hit the rule later, you will be in trouble.

**Comment by membphis** [comment]
@ShiningRush I remember we discussed this method before, you can look at this PR

**Comment by membphis** [comment]
this way is much better than the old way ^_^

File: apisix/core/schema.lua [hunk 1]
```diff
@@ -20,7 +20,11 @@ local lrucache = require("apisix.core.lrucache")
 local cached_validator = lrucache.new({count = 1000, ttl = 0})
 local pcall = pcall
 
-local _M = {version = 0.3}
+local _M = {
+    version = 0.3,
+    TYPE_ANY = 1,
```
Comment (ShiningRush) [inline apisix/core/schema.lua:25]: - `TYPE_ANY` is confused, I think `TYPE_CONF` or `TYPE_OPT` would be better, how do you think?
- It seems like the code of checking plugin's config does not using this parameter, are there any special considerations?

**Review by ShiningRush — CHANGES_REQUESTED** [review]

File: apisix/admin/plugins.lua [hunk 2]
```diff
@@ -29,10 +30,16 @@ local table_insert = table.insert
 local _M = {}
 
 
-function _M.check_schema(plugins_conf)
+function _M.check_schema(plugins_conf, schema_type)
     for name, plugin_conf in pairs(plugins_conf) do
         core.log.info("check plugin scheme, name: ", name, ", configurations: ",
                       core.json.delay_encode(plugin_conf, true))
+        if type(plugin_conf) ~= "table" then
```
Comment (ShiningRush) [inline apisix/admin/plugins.lua:37]: No test case cover this logic.

**Review by ShiningRush — CHANGES_REQUESTED** [review]

File: apisix/admin/plugins.lua (hunk 2 above)
Comment (spacewander) [reply apisix/admin/plugins.lua:37]: @ShiningRush 
Updated.

**Review by ShiningRush — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Pull request merged**